From f7c0da499291b88eb904388133ec8ff08f39a309 Mon Sep 17 00:00:00 2001
From: Bryce Harrington <bryce@bryceharrington.org>
Date: Tue, 10 Feb 2009 14:03:52 -0800
Subject: [PATCH] Default ModeDebug to On.

 This patch is intended to be carried during Ubuntu development only.
 It should be reverted prior to -beta.

Signed-off-by: Bryce Harrington <bryce@bryceharrington.org>
---
 hw/xfree86/doc/man/xorg.conf.man.pre |    2 +-
 hw/xfree86/modes/xf86Crtc.c          |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/hw/xfree86/doc/man/xorg.conf.man.pre b/hw/xfree86/doc/man/xorg.conf.man.pre
index d9a4b35..201dfc3 100644
--- a/hw/xfree86/doc/man/xorg.conf.man.pre
+++ b/hw/xfree86/doc/man/xorg.conf.man.pre
@@ -1087,7 +1087,7 @@ This is rarely used.
 .TP 7
 .BI "Option \*qModeDebug\*q \*q" boolean \*q
 Enable printing of additional debugging information about modesetting to
-the server log.
+the server log.  If this option is omitted, it defaults to "true".
 .ig
 .TP 7
 This optional entry allows an IRQ number to be specified.
diff --git a/hw/xfree86/modes/xf86Crtc.c b/hw/xfree86/modes/xf86Crtc.c
index b972974..f7e5066 100644
--- a/hw/xfree86/modes/xf86Crtc.c
+++ b/hw/xfree86/modes/xf86Crtc.c
@@ -2186,7 +2186,7 @@ xf86InitialConfiguration (ScrnInfoPtr scrn, Bool canGrow)
 			scrn->options,
 			config->options);
     config->debug_modes = xf86ReturnOptValBool (config->options,
-						OPTION_MODEDEBUG, FALSE);
+						OPTION_MODEDEBUG, TRUE);
 
     if (scrn->display->virtualX)
 	width = scrn->display->virtualX;
-- 
1.6.0.4

